snetd/cmd/channel.go

```diff
@@ -0,0 +1,76 @@
+package cmd
+
+import (
+	"fmt"
+	"github.com/singnet/snet-daemon/escrow"
+	"github.com/singnet/snet-daemon/etcddb"
+	"github.com/spf13/cobra"
+	"math/big"
+)
+
+// ListChannelsCmd shows list of channels from shared storage
+var ChannelCmd = &cobra.Command{
+	Use:   "channel",
+	Short: "Manage operations on payment channels",
+	Long: "allows us to perform operations on channels with given channel ID." +
+		" User can use 'snetd channel --unlock {channelID}' command to unlock the channel manually.",
+	RunE: func(cmd *cobra.Command, args []string) error {
+		return RunAndCleanup(cmd, args, newChannelCommand)
+	},
+}
+
+//Channel command type
+type channelCommand struct {
+	etcdclient       *etcddb.EtcdClient
+	paymentChannelId *big.Int
+}
+
+// initializes and returns the new channel command object
+func newChannelCommand(cmd *cobra.Command, args []string, components *Components) (command Command, err error) {
+	channelId, err := getPaymentChannelId(cmd)
+	if err != nil {
+		return
+	}
+	command = &channelCommand{
+		etcdclient:       components.EtcdClient(),
+		paymentChannelId: channelId,
+	}
+	return
+}
+
+func getPaymentChannelId(cmd *cobra.Command) (id *big.Int, err error) {
+	if paymentChannelId == "" {
+		return nil, nil
+	}
+	value := &big.Int{}
+	err = value.UnmarshalText([]byte(paymentChannelId))
+	if err != nil {
+		return nil, fmt.Errorf("Incorrect decimal number format: %v, error: %v", paymentChannelId, err)
+	}
+	return value, nil
+}
+
+// command's run method
+func (command *channelCommand) Run() (err error) {
+	if command.paymentChannelId == nil {
+		return fmt.Errorf("--unlock channel-id must be set")
+	}
+	if command.paymentChannelId != nil {
+		return command.unlockChannel()
+	}
+	return
+}
+
+// unlocks the channel with a given channel ID
+func (command *channelCommand) unlockChannel() (err error) {
```

Comment:
I would re-use same `PrefixedAtomicStorage` which is used in `EtcdLocker` see https://github.com/singnet/snet-daemon/blob/05b4021d74e0516191289d3b95f68d1f94132385/escrow/lock.go#L20-L27
otherwise you have a risk that someone changes the locker code and will not change yours code and will break the functionality. 

I would do the following: 
- introduce new component etcdLockerStorage
- initialize it as in etcdLocker constructor: https://github.com/singnet/snet-daemon/blob/05b4021d74e0516191289d3b95f68d1f94132385/escrow/lock.go#L22-L25
- use this component in `NewEtcdLocker` and in `channelComand` (instead of etcdClient).
- use `PrefixedAtomicStorage.Delete()` method to remove the key

Reply:
made the changes as discussed

Reply:
@Dinesh2521, thank you looks perfect to me, just rename the component and constructor to something less abstract, thanks.